tests/devices/qutrit_mixed/test_qutrit_mixed_simulate.py

```diff
@@ -0,0 +1,735 @@
+
+
+def expected_TRX_circ_expval_jacobians(phi, subspace):
+    """Find the Jacobians of expect-values of GellManns 2,3,5,8
+    on a circuit with a TRX gate"""
+    if subspace == (0, 1):
+        gellmann_2 = -np.cos(phi)
+        gellmann_3 = -np.sin(phi)
+        gellmann_5 = 0
+        gellmann_8 = 0
+    if subspace == (0, 2):
+        gellmann_2 = 0
+        gellmann_3 = -np.sin(phi) / 2
+        gellmann_5 = -np.cos(phi)
+        gellmann_8 = np.sqrt(1 / 3) * -(1.5 * np.sin(phi))
+    return np.array([gellmann_2, gellmann_3, gellmann_5, gellmann_8])
+
+
+def expected_TRX_circ_state(phi, subspace):
+    """Find the state after applying TRX gate on |0>"""
+    expected_vector = np.array([0, 0, 0], dtype=complex)
+    expected_vector[subspace[0]] = np.cos(phi / 2)
+    expected_vector[subspace[1]] = -1j * np.sin(phi / 2)
+    return np.outer(expected_vector, np.conj(expected_vector))
+
+
+class TestCurrentlyUnsupportedCases:
+    """Test currently unsupported cases, such as sampling expval or samples without shots"""
+
+    # pylint: disable=too-few-public-methods
+    def test_sample_based_observable(self):
+        """Test sample-only measurements raise a NotImplementedError."""
+
+        qs = qml.tape.QuantumScript(measurements=[qml.sample(wires=0)])
+        with pytest.raises(NotImplementedError):
+            simulate(qs)
+
+    @pytest.mark.parametrize("mp", [qml.probs(0), qml.probs(op=qml.GellMann(0, 2))])
+    def test_invalid_samples(self, mp):
+        """Test Sampling MeasurementProcesses that are currently unsupported on this device"""
+        qs = qml.tape.QuantumScript(ops=[qml.TAdd(wires=(0, 1))], measurements=[mp], shots=10)
+        with pytest.raises(NotImplementedError):
+            simulate(qs)
+
+
+def test_custom_operation():
+    """Test execution works with a manually defined operator if it has a matrix."""
+
+    # pylint: disable=too-few-public-methods
+    class MyOperator(qml.operation.Operator):
+        num_wires = 1
+
+        @staticmethod
+        def compute_matrix():
+            return np.array([[0, 0, 1], [0, 1, 0], [1, 0, 0]])
+
+    qs = qml.tape.QuantumScript([MyOperator(0)], [qml.expval(qml.GellMann(0, 8))])
+
+    result = simulate(qs)
+    assert qml.math.allclose(result, -np.sqrt(4 / 3))
+
+
+# pylint: disable=too-few-public-methods
+class TestStatePrepBase:
+    """Tests integration with various state prep methods."""
+
+    def test_basis_state(self):
+        """Test that the BasisState operator prepares the desired state."""
+        qs = qml.tape.QuantumScript(
+            ops=[qml.QutritBasisState((2, 1), wires=(0, 1))],
+            measurements=[qml.probs(wires=(0, 1, 2))],
+        )
+        probs = simulate(qs)
+        expected = np.zeros(27)
+        expected[21] = 1.0
+        assert qml.math.allclose(probs, expected)
+
+
+class TestBasicCircuit:
+    """Tests a basic circuit with one rx gate and two simple expectation values."""
```

Comment:
```suggestion
    """Tests a basic circuit with one TRX gate and two simple expectation values."""
```

Reply:
More than two expvals are being returned in the script creation function

Reply:
agreed, I fixed the comment